packages/nimble-components/src/table/index.ts

```diff
@@ -268,19 +271,55 @@ export class Table<
         );
     }
 
+    private updateTanStack(): void {
+        const updatedOptions: Partial<TanStackTableOptionsResolved<TData>> = {
+            state: {}
+        };
+
+        if (this.updateTracker.updateColumnSort) {
+            updatedOptions.state!.sorting = this.getTanStackSortState();
+        }
+        if (this.updateTracker.updateColumnDefinition) {
+            updatedOptions.columns = this.getTanStackColumns();
+        }
+        if (this.updateTracker.updateRowIds) {
+            updatedOptions.getRowId = this.getTanStackRowIdFunction();
+        }
+        if (this.updateTracker.requiresTanStackDataReset) {
+            // Perform a shallow copy of the data to trigger tanstack to regenerate the row models and columns.
+            updatedOptions.data = [...this.table.options.data];
+        }
+
+        this.updateTableOptions(updatedOptions);
+    }
+
+    private updateActionMenuSlots(): void {
+        const slots = new Set<string>();
+        for (const column of this.columns) {
+            if (column.actionMenuSlot) {
+                slots.add(column.actionMenuSlot);
+            }
+        }
+        this.actionMenuSlots = Array.from(slots);
+    }
+
+    private validate(data: TData[]): void {
```

Comment:
Would `data` here ever not be `this.table.options.data`?

Reply:
Nope, not with the existing code. I'll change this to use `this.table.options.data` within `validate()` rather than pass it in.